cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/TopicLeaderDistributionGoal.java

```diff
@@ -343,11 +350,10 @@ private static boolean skipBrokerRebalance(Broker broker,
                                              ClusterModel clusterModel,
                                              Collection<Replica> replicas,
                                              boolean requireLessReplicas,
-                                             boolean requireMoreReplicas,
                                              boolean hasOfflineTopicReplicas,
                                              boolean moveImmigrantReplicaOnly) {
```

Comment:
Hm, what's the purpose of the changes to this function?

Reply:
Previously the goal did two kinds of rebalances: for brokers that led too many partitions it moved the leadership "out", and for brokers that led too few it moved the leadership "in". I thought that we actually don't need both, it's enough to fix the balance for brokers that have lead too many partitions, and that would automatically fix the balance for brokers that have too few.

Actually, now I realized that when the ideal balance is not achievable, not doing two-way balancing actually may leave some brokers leading fewer partitions and they violate the lower boundary condition, and in general the overall distribution of the load between brokers may be less even than it could be. I'll revert this part of the change and will do two-way rebalancing, in particular doing replica swaps in both cases two.

Reply:
Thanks, that sounds reasonable.